### DISCUSSION TITLE
Pricing Speech-To-Text by stereo audio (2 channels)

### DISCUSSION THREAD

**Discussion by Jguarniz04** [deepgram/community#932] [General help] [answered]
I have been reading about your speech to text tool and I have been trying it, but I have a question, if I transcribe a stereo audio with 2 channels lasting 1 minute, in a single request, will the cost of this be only for 1 minute? I see that they separate each speaker and make the multi-channel differentiation, but I want to know if the cost of this would be only for 1 minute, or is the cost transferred to each channel and would be 2 minutes? 1 for each channel, this is not very clear to me in the documentation.
I would appreciate it if you could clear up that doubt for me.

**Comment by team-deepgram** [comment]
Thanks for asking your question about Deepgram! If you didn't already include it in your post, please be sure to add as much detail as possible so we can assist you efficiently, such as:
- The `request_id` if you have a question about your requests or transcription responses.
- The features you used or the full `api.deepgram.com` URL you sent your request to, including parameters.
- Any code snippets you can share.

**Comment by jjmaldonis** [comment]
Hi @Jguarniz04, the cost for multichannel is applied for each channel. So if you have a two channel file that is 1 minute long and transcribe it with `multichannel=true`, you will be charged for 2 minutes of audio processing. This is because Deepgram is transcribing 2x the amount of audio and performing 2x the computation.

**Reply by Jguarniz04** [reply]
Hi @jjmaldonis thank you for the answer to my question.

After searching, I found a section where I can see the test transcription that I made to a 1:44 stereo audio, but it shows me the cost for the total duration of the audio, without duplicating the duration for each channel as mentioned.
Since the transcription of both channels was done correctly.
So what Deepgram charged is wrong in this case?
because according to what was explained, the cost should be 0.01505.

I appreciate the clarification on this, since I need to be clear about this case. Thank you.
![image](https://github.com/user-attachments/assets/a3e61a3f-640d-4ccf-8f13-6699d0bf65a5)

**Reply by jjmaldonis** [reply, marked as answer]
I looked up the request ID in the screenshot (`7f13b813-37e1-462c-a61e-67d986548a57`) in our logs, and the following query parameters were used:

```
{
  "model": "nova-2",
  "language": "es-419",
  "diarize": "true",
  "smart_format": "true",
  "paragraphs": "true"
}
```
`?language=es-419&model=nova-2&smart_format=true&paragraphs=true&diarize=true`

So for this request you used diarization rather than `multichannel=true`.

**Reply by Jguarniz04** [reply]
I understand, thank you very much for the clarification, I started with this recently and I am still learning how to use the service.
If you have any example code, that I can run in python, for the analysis of a stereo audio loaded locally from my PC, I would greatly appreciate it if you shared it with me.

**Reply by petro-sam** [reply]
> Hi @Jguarniz04, the cost for multichannel is applied for each channel. So if you have a two channel file that is 1 minute long and transcribe it with `multichannel=true`, you will be charged for 2 minutes of audio processing. This is because Deepgram is transcribing 2x the amount of audio and performing 2x the computation.

Why DeepGram can't skip "silent" frames while transcrbing to avoid extra computations and to allow us optimal API usage? I think it would be quite simple to fix. Current implementation of multichannel is useless due to overpay. ATM we have to open new websocket each time new user start speaking.
